### PR TITLE
Bug1413156 - lodash to ES6: replace _.reduce, _.assign, _.flatten, _.…

### PR DESCRIPTION
**Pull request by Dottori**
…isArray

Bug1413156 - lodash to ES6: replace _.reduce, _.assign, _.flatten, _.isArray

Bug1413156 - lodash to ES6: replace _.reduce, _.assign, _.flatten, _.isArray

@camd @edmorley  (also includes instance of _.every in bugfiler file)

### CODE REVIEW

File: ui/js/controllers/filters.js [hunk 1]
```diff
@@ -51,12 +50,12 @@ treeherderApp.controller('JobFilterCtrl', [
...
 
         $scope.isFilterOn = function (filter) {
-            if (_.keys($scope.filterGroups).indexOf(filter) !== -1) {
-                // this is a filter grouping, so toggle all on/off
-                return _.some(
-                    _.at($scope.resultStatusFilters,
-                    $scope.filterGroups[filter].resultStatuses)
-                );
+            if (Object.keys($scope.filterGroups).indexOf(filter) !== -1) {
+                const filterOn = [];
```
Comment (camd) [inline ui/js/controllers/filters.js]: It seems like we could use a ``reduce`` insead of this ``for .. of`` statement.  Would that work?

**Comment by Dottori** [comment]
Hi @camd I used .map to replace the for loop =). Let me know if the changes are okay or if you would prefer .reduce =D.

**Comment by camd** [comment]
Hi @Dottori - I tried these changes and unfortunately, they aren't working.  If you run TH locally with ``yarn start`` and check the toolbar when no filters are set (the default) then all the colored dots should be filled, but the ones that this effects (failures and in-progress) aren't filled.  Would you take another crack at this?  Thanks!  :)

**Comment by Dottori** [comment]
=). No worries. Thank you again for taking the time to review the PR and sorry for missing the return the first time around. Careless errors =P.

**Pull request merged**